madgraph/core/diagram_generation.py

```diff
@@ -1730,11 +1730,25 @@ def generate_multi_amplitudes(cls,process_definition,
         except KeyError:
             fstags = []
 
+        try:
+            istags = [leg['is_tagged'] for leg in process_definition['legs'] \
+                 if 'is_tagged' in leg.keys() and leg['state'] == False]
+
+        except KeyError:
+            istags = []
+
```

Comment:
This sounds a quite dangerous error catching, maybe this one is better (depending which error you want to catch)
```suggestion
        try:
            process_definition['legs']
        except KeyError:
            istags = [] 
        else:
            istags = [leg['is_tagged'] for leg in process_definition['legs'] \
                 if 'is_tagged' in leg.keys() and leg['state'] == False]
```

Reply:
Do you agree with this suggestion?
If you do then I can do the merge of this branch

Reply:
Sorry, I forgot to check this comment. To be honest, I cannot remember the KeyError I was trying to catch. I think it was related to the key 'is_tagged', but I need to check this again.

Reply:
This is exactly the reason why those except need to be "clear" otherwise it is difficult to maintain.

A potential missing "is_tagged" should be handled via the 
```
if 'is_tagged' in leg.keys()
```
so in that case maybe even better to do
```suggestion
        istags = [leg['is_tagged'] for leg in process_definition['legs'] \
                 if 'is_tagged' in leg.keys() and leg['state'] == False]
```

Reply:
I agree with your latest solution, it works well, and I don’t see any need for the `try-except` construction. I probably just copied it from the previous lines, where the same pattern is used for tagged final-state particles. Maybe @marcozaro remembers why it was implemented that way; I don’t see the need for it there either.